### PR TITLE
Oe/86bz12tn4/realization stepper v3

### PR DESCRIPTION
**Pull request by guardian-001**
- realization stepper 
- hook extraction
- login validation message fix 
- translation update
- api services

## note ! : i fixed all comments of pr v1 and pr v2 in this pr

### CODE REVIEW

File: src/modules/architect/realization/steps/adopted-style.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    isError,
+    error,
+    onSubmit,
+    handleSubmit,
+    control,
+    onHandleBack,
+  } = useStyles();
+  return (
+    <View className="mb-5 flex h-full w-full flex-1 items-center justify-between gap-6  ">
+      {isError ? (
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/realization/steps/adopted-style.tsx:24]: try to not use ? : for more code readablity use && instead

File: src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const { formData } = useFormStepper<SignupFormDataType>();
+  const signup = useSignupApi();
+  const router = useRouter();
+  const [timezone] = useTimezone();
+  const handleConfirmationStep = () => {
+    signup.mutate(formData, {
+      onSuccess: () => {
+        router.push('(architect)/(public)/login');
+      },
+      onError: () => {},
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx:19]: remove on error and pass the the router in the mutation parameter

File: src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import { useSignupApi } from '@/api/auth';
+import { useTimezone } from '@/core';
+import { useFormStepper } from '@/shared';
+
+import type { SignupFormDataType } from '../types';
+
+export const useDemoConfirmation = () => {
+  const { formData } = useFormStepper<SignupFormDataType>();
+  const signup = useSignupApi();
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx:11]: change the nampe API to mutation in this case for all mutations that u have in the project

File: src/modules/architect/signup/shared/hooks/use-profile.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    firstName: formData?.firstName,
+    lastName: formData?.lastName,
+    email: formData?.email,
+    phoneNumber: formData?.phoneNumber,
+    address: formData?.address,
+    architectIdentifier: formData?.architectIdentifier,
+  });
+
+  const onSubmit = (data: CreateAccountFormType) => {
+    setFormData((prev: any) => ({
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/signup/shared/hooks/use-profile.tsx:29]: u solved it in the previous one and u left it here, change the any if u have the formData type

File: src/modules/architect/signup/shared/hooks/use-speciality.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      architectSpeciality: formData.architectSpeciality,
+    }
+  );
+
+  const onHandleBack = () => {
+    route.push('/(architect)/(public)/login');
+  };
+
+  const onSubmit = (data: SpecialityFormType) => {
+    setFormData((prev: any) => ({
```
Comment (Ghazi-Chaftar) [inline src/modules/architect/signup/shared/hooks/use-speciality.tsx:28]: here too

File: src/modules/login/login-form.tsx [hunk 6]
```diff
@@ -25,17 +25,22 @@ export const LoginForm = () => {
   const space = useRouteName();
   const signIn = useAuth.use.signIn();
   const [checked, setChecked] = useState(true);
+  const [errors, setErrors] = useState('');
   const { handleSubmit, control } = useCustomForm(LoginFormSchema);
   const login = useLoginApi();
 
   const onSubmit = (data: LoginFormType) => {
     login.mutate(data, {
       onSuccess: (response) => {
```
Comment (Ghazi-Chaftar) [inline src/modules/login/login-form.tsx:34]: always the onsuccess and on error should be defined in the mutation before using it, to keep ur logic outside ur ui

File: src/shared/components/calendar.tsx [hunk 7]
```diff
@@ -102,8 +102,8 @@ export const Calendar = () => {
       {errors?.date && <Text>{errors.date.message}</Text>}
       {errors?.timeSlot && <Text>{errors.timeSlot.message}</Text>}
       <StepButtons
-        previous={{ handlePreviousStep: onHandleBack, label: 'signup.retour' }}
-        next={{ handleSubmit: handleSubmit(onSubmit), label: 'signup.suivant' }}
+        previous={{ handlePreviousStep: onHandleBack, label: 'common.retour' }}
```
Comment (Ghazi-Chaftar) [inline src/shared/components/calendar.tsx:105]: retour et suivant change them to back and next (in code we only use english)

File: src/translations/ar/common.json [hunk 8]
```diff
@@ -4,5 +4,10 @@
   "choosePassword": "اختر كلمة المرور",
   "enterPassword": "أدخل كلمة المرور",
   "confirmPassword": "تأكيد كلمة المرور",
-  "start": "بدأ"
+  "start": "بدأ",
+  "ignorer": "تخطي",
```
Comment (Ghazi-Chaftar) [inline src/translations/ar/common.json:8]: here too

**Review by Ghazi-Chaftar — CHANGES_REQUESTED** [review]
left some comments

File: src/app/(supplier)/(public)/login.tsx [hunk 9]
```diff
@@ -13,11 +12,14 @@ export type LoginFormSupplierProps = {
 };
 export default function Login() {
   const router = useRouter();
-  const signIn = useAuth.use.signIn();
+  // const signIn = useAuth.use.signIn();
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/app/(supplier)/(public)/login.tsx:15]: removed

File: src/api/architect/project/use-create-project.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+type Response = AxiosResponse;
+type Variables = FormData;
+
+export const useCreateProjectApi = createMutation<
+  Response,
+  Variables,
+  AxiosError
+>({
+  mutationKey: ['createProject'],
+  mutationFn: async (formData: FormData) => {
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/api/architect/project/use-create-project.tsx:15]: done

File: src/app/(supplier)/(public)/login.tsx [hunk 11]
```diff
@@ -13,11 +12,14 @@ export type LoginFormSupplierProps = {
...
   useSoftKeyboardEffect();
 
   const onSubmit: LoginFormSupplierProps['onSubmit'] = () => {
-    signIn({ access: 'access-token', refresh: 'refresh-token' });
+    // signIn({
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/app/(supplier)/(public)/login.tsx:19]: removed

File: src/modules/architect/realization/shared/components/close-btn.tsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+};
+export default function CloseBtn({ handleAction }: closeProps) {
+  return (
+    <View className="absolute right-1 top-2 z-10 flex h-8 w-8 items-center justify-center rounded-full bg-slate-300/70">
+      <TouchableOpacity
+        className="  flex h-8 w-8 items-center justify-center  "
+        onPress={handleAction}
+      >
+        <Text>X</Text>
+        {/* <Close color={colors.blue} /> */}
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/shared/components/close-btn.tsx:14]: removed

File: src/modules/architect/realization/shared/components/general-infos.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+            {categoryName}
+          </Text>
+        </View>
+        <View className="flex flex-1 justify-start">
+          <Text
+            tx={'realisation.finalStep.localisation'}
+            className="mb-2 text-start text-sm  font-extrabold"
+          />
+          <Text className="mb-2 text-start text-sm font-extrabold  text-description">
+            {formData.city}
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/shared/components/general-infos.tsx:27]: fixed

File: src/modules/architect/realization/shared/components/images-list.tsx [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+                  handleAction={() =>
+                    removeImage(
+                      index > 0 ? index + 1 + subIndex : index + subIndex
+                    )
+                  }
+                />
+                <Pressable
+                  onPress={() =>
+                    handleImagePress(
+                      index > 0 ? index + 1 + subIndex : index + subIndex
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/shared/components/images-list.tsx:46]: fixed, moved under the function handle image press

File: src/modules/architect/realization/shared/hooks/use-image-pairs.tsx [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import type { ImageInfo } from '../types';
+
+export const useImagePairs = (images: ImageInfo[]) => {
+  const groupImagesInPairs = () => {
+    const pairs = [];
+    for (let i = 0; i < images.length; i += 2) {
```
[the remark this answers is not in the document]
Comment (guardian-001) [reply src/modules/architect/realization/shared/hooks/use-image-pairs.tsx:6]: fixed

File: src/modules/architect/realization/steps/adopted-style.tsx (hunk 1 above)
Comment (guardian-001) [reply src/modules/architect/realization/steps/adopted-style.tsx:24]: i need the else property here, it's better to use ?

File: src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx (hunk 2 above)
Comment (guardian-001) [reply src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx:19]: fixed

File: src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx (hunk 3 above)
Comment (guardian-001) [reply src/modules/architect/signup/shared/hooks/use-demo-confirmation.tsx:11]: megdich tells me to use Api

File: src/modules/architect/signup/shared/hooks/use-profile.tsx (hunk 4 above)
Comment (guardian-001) [reply src/modules/architect/signup/shared/hooks/use-profile.tsx:29]: done

File: src/modules/architect/signup/shared/hooks/use-speciality.tsx (hunk 5 above)
Comment (guardian-001) [reply src/modules/architect/signup/shared/hooks/use-speciality.tsx:28]: done here too

File: src/modules/login/login-form.tsx (hunk 6 above)
Comment (guardian-001) [reply src/modules/login/login-form.tsx:34]: done

File: src/shared/components/calendar.tsx (hunk 7 above)
Comment (guardian-001) [reply src/shared/components/calendar.tsx:105]: done

File: src/translations/ar/common.json (hunk 8 above)
Comment (guardian-001) [reply src/translations/ar/common.json:8]: done all

**Pull request merged**